app/models/event_schedule.rb

```diff
@@ -37,6 +39,16 @@ def replacement?
 
   private
 
+  def start_after_end_hour
+    return unless event && start_time && event.program && event.program.conference && event.program.conference.end_hour
```

Comment:
there is a validation that ensure that `event ` and `start_time` are present, so it is not needed to check it here. And the same for `program` in Event. So:

``` Ruby
return unless event.program.conference.try(:end_hour)